### PR TITLE
Change fly_camera example to use winit's DeviceEvent

### LINKED ISSUE #636: Fly Camera rotation sensitivity tied to FPS
This is occurring on Windows 10, I am not sure about other operating systems.

It appears that the Fly Camera rotation speed is dependent upon the frame rate, if I run the fly_camera example with:

    cargo run --release --example fly_camera

The camera moves way too fast to even control with the mouse.

If I enable vsync or enable a `FrameRateLimitStrategy` with a reasonable FPS the rotation becomes much slower and more manageable, I think that regardless of FPS the rotation speed should be the same.

### PR DESCRIPTION
**Pull request by MrMinimal**
Following #699 this PR focuses on implementing raw mouse input for the fly_camera example.
It replaces the mouse input relative to cursor position with raw mouse input (which also sends values even if the cursor does not move).
This might fix #636 but I'd have to test that.

Also the mouse cursor is not forcibly centered every frame, but instead winit's `CursorState::Grab` is used.

I have only tested this on Windows 10 so far, not sure if the movement is still framerate dependent.

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/705)

### CODE REVIEW

File: amethyst_controls/src/systems.rs [hunk 1]
```diff
@@ -82,11 +81,13 @@ where
 
 /// The system that manages the view rotation.
 /// Controlled by the mouse.
+/// Handles raw mouse input.
 pub struct FreeRotationSystem<A, B> {
     sensitivity_x: f32,
     sensitivity_y: f32,
     _marker1: PhantomData<A>,
     _marker2: PhantomData<B>,
+    event_reader: Option<ReaderId<Event>>,
```
Comment (torkleyy) [inline amethyst_controls/src/systems.rs]: This will need a `setup` method in the `System`

Comment (torkleyy) [reply amethyst_controls/src/systems.rs]: Ah sorry, the diff confused me.

**Comment by MrMinimal** [comment]
Seems to not work too well on my Ubuntu 18.04 with X11. Will investigate :mag: 
```
[amethyst::app][INFO] Initializing Amethyst...
[amethyst::app][INFO] Version: v0.5.1-723-g3c9e8e7
[amethyst::app][INFO] Platform: x86_64-unknown-linux-gnu
[amethyst::app][INFO] Amethyst git commit: 3c9e8e779830cdf95f47dd2224eb861a6654f5e6
[amethyst::app][INFO] Rustc version: 1.27.0-nightly Nightly
[amethyst::app][INFO] Rustc git commit: 428ea5f6b9e7c6e5ee3294fe9f105e77e89ab407
[xcb] Unknown sequence number while processing queue
[xcb] Most likely this is a multi-threaded client and XInitThreads has not been called
[xcb] Aborting, sorry about that.
fly_camera: ../../src/xcb_io.c:259: poll_for_event: Assertion `!xcb_xlib_threads_sequence_lost' failed.
Aborted (core dumped)
```

**Comment by Rhuagh** [comment]
Conflicts need to be fixed before merge.

**Comment by MrMinimal** [comment]
@Rhuagh Well, I'll fix this tomorrow, the camera behaves wierdly with the current amethyst/develop branch. Currently broken.

**Comment by MrMinimal** [comment]
Wrong button, but up to date with current `amethyst/develop` now

File: amethyst_controls/src/systems.rs [hunk 2]
```diff
@@ -106,87 +108,68 @@ where
...
-                    error!("Unable to set the cursor position! Error: {:?}", err);
-                }
-            });
-        } else {
-            release_cursor(&mut msg);
-        }
-    }
+    fn run(&mut self, (events, mut transform, tag): Self::SystemData) {
+        for event in events.read(&mut self.event_reader.as_mut().unwrap()) {
+            match event {
```
Comment (torkleyy) [inline amethyst_controls/src/systems.rs]: should match on *event

File: amethyst_controls/src/systems.rs [hunk 3]
```diff
@@ -106,87 +108,68 @@ where
...
+                &Event::DeviceEvent { ref event, .. } => {
+                    match event {
+                        &DeviceEvent::MouseMotion  { delta: (x, y) } => {
```
Comment (torkleyy) [inline amethyst_controls/src/systems.rs]: same here, try clippy

File: amethyst_controls/src/resources.rs [hunk 4]
```diff
@@ -1,12 +0,0 @@
-/// Struct which holds information about whether the window is focused.
-/// Written to by MouseFocusUpdateSystem
-#[derive(Default)]
-pub struct WindowFocus {
```
Comment (jojolepro) [inline amethyst_controls/src/resources.rs]: This is useful for other use-cases, please don't remove it.

File: amethyst_controls/src/systems.rs [hunk 5]
```diff
@@ -106,87 +108,68 @@ where
...
-                    transform.pitch_local(Deg(offset_y * self.sensitivity_y));
-                    transform.yaw_global(Deg(offset_x * self.sensitivity_x));
-                }
-            }
-        }
-    }
-}
-
-/// The system that locks the mouse to the center of the screen. Useful for first person camera.
-pub struct MouseCenterLockSystem;
```
Comment (jojolepro) [inline amethyst_controls/src/systems.rs]: Please keep it, it can still be useful too.

File: amethyst_controls/src/resources.rs (hunk 4 above)
Comment (Rhuagh) [reply amethyst_controls/src/resources.rs]: Agreed.

File: amethyst_controls/src/systems.rs (hunk 5 above)
Comment (Rhuagh) [reply amethyst_controls/src/systems.rs]: Can you give an example for when this would be useful over using `Grab`?

Comment (jojolepro) [reply amethyst_controls/src/systems.rs]: 1) As a fallback from the grab method for other pointer devices.
2) If for some reason you actually want to force the mouse to be centered as a game feature.

**Comment by torkleyy** [comment]
https://travis-ci.org/amethyst/amethyst/jobs/380586984#L344

```
error[E0432]: unresolved import `amethyst::controls::ArcBallControlTag`
 --> examples/arc_ball_camera/main.rs:7:26
  |
7 | use amethyst::controls::{ArcBallControlTag, ArcBallMovementSystem, FlyControlTag,
  |                          ^^^^^^^^^^^^^^^^^ no `ArcBallControlTag` in `controls`
error[E0432]: unresolved import `amethyst::controls::ArcBallMovementSystem`
 --> examples/arc_ball_camera/main.rs:7:45
  |
7 | use amethyst::controls::{ArcBallControlTag, ArcBallMovementSystem, FlyControlTag,
  |                                             ^^^^^^^^^^^^^^^^^^^^^ no `ArcBallMovementSystem` in `controls`. Did you mean to use `FlyMovementSystem`?
error[E0432]: unresolved import `amethyst::controls::MouseCenterLockSystem`
 --> examples/arc_ball_camera/main.rs:8:46
  |
8 |                          FreeRotationSystem, MouseCenterLockSystem, MouseFocusUpdateSystem};
  |                                              ^^^^^^^^^^^^^^^^^^^^^ no `MouseCenterLockSystem` in `controls`
```

**Comment by MrMinimal** [comment]
Pretty sure I didn't click that... still wrangling with git

**Comment by jojolepro** [comment]
Will need a new commit to reopen by the looks of it.

**Pull request closed without merging**